src/core/Song.cpp

```diff
@@ -1515,36 +1514,31 @@ void Song::removeController( Controller * controller )
 
 void Song::clearErrors()
 {
-	m_errors->clear();
+	m_errors.clear();
 }
 
 
 
 void Song::collectError( const QString error )
 {
-	m_errors->append( error );
+	m_errors.append( error );
 }
 
 
 
 bool Song::hasErrors()
 {
-	return ( m_errors->length() > 0 );
+	return ( m_errors.length() > 0 );
 }
 
 
 
-QString* Song::errorSummary()
+QString Song::errorSummary()
 {
-	QString* errors = new QString();
+	QString errors = m_errors.join('\n') + '\n';
```

Comment:
Constructor of QString expects a const char* here. Change the '\n's to "\n" to make them string literals.

Reply:
Yup, although It's the `join` function that expects `const char*`, not the constructor. Appears to be a Qt5-only feature.

Reply:
Fixed via 670ca4f